packages/autocertifier/test/end-to-end/clientServer.test.ts

```diff
@@ -0,0 +1,175 @@
+import { 
+    ConnectionManager,
+    ListeningRpcCommunicator,
+    NodeType,
+    PeerDescriptor,
+    PeerID,
+    Simulator,
+    SimulatorTransport,
+    DefaultConnectorFacade
+} from '@streamr/dht'
+import { AutoCertifierClient, createSelfSignedCertificate } from '@streamr/autocertifier-client'
+import os from 'os'
+import fs from 'fs'
+import { RestServer } from '../../src/RestServer'
+import { CertifiedSubdomain } from '@streamr/autocertifier-client'
+import { Session } from '@streamr/autocertifier-client'
+import { v4 } from 'uuid'
+import { Logger } from '@streamr/utils'
+import { StreamrChallenger } from '../../src/StreamrChallenger'
+import { SessionIdRequest, SessionIdResponse } from '../../src/proto/packages/autocertifier/protos/AutoCertifier'
+
+const logger = new Logger(module)
+
+let server: RestServer
+const dir = os.tmpdir()
+let certifiedSubdomain: CertifiedSubdomain
+
+const createTestSubdomain = (validityMillis?: number) => {
+    if (validityMillis) {
+        const fakeCerts = createSelfSignedCertificate('localhost', 1200)
+        certifiedSubdomain = { 
+            fqdn: 'localhost',
+            subdomain: 'fwefwafeaw',
```

Comment:
Is this test data or some specific subdomain? Could use 'mock-subdomain' or similar if it is just mock data